src/rdiff_backup/SetConnections.py

```diff
@@ -210,11 +210,32 @@ def _parse_file_desc(file_desc):
 
 
 def _fill_schema(host_info):
-    """Fills host_info into the schema and returns remote command"""
+    """
+    Fills host_info and optionally the version into the schema and returns remote command.
+    """
+    assert isinstance(host_info, bytes), (
+        "host_info parameter must be bytes not {thi}.".format(
+            thi=type(host_info)))
     try:
-        return __cmd_schema % host_info
-    except TypeError:
-        Log.FatalError("Invalid remote schema:\n\n%s\n" % _safe_str(__cmd_schema))
+        # for security reasons, we accept only specific format placeholders
+        # h for host_info, vx,vy,vz for version x.y.z
+        # and the host placeholder is mandatory
+        if ((re.findall(b"{[^}]*}", __cmd_schema)
+             != re.findall(b"{h}|{v[xyz]}", __cmd_schema))
+                or (b"{h}" not in __cmd_schema
+                    and b"%s" not in __cmd_schema)):  # compat200
+            raise KeyError
+        if b"{h}" in __cmd_schema:
+            ver_split = Globals.version.split(".")
+            # bytes doesn't have a format method, hence the conversions
+            return os.fsencode(os.fsdecode(__cmd_schema).format(
+                h=os.fsdecode(host_info),
+                vx=ver_split[0], vy=ver_split[1], vz=ver_split[2]))
+        else:  # compat200: accepts "%s" as host place-holder
+            return __cmd_schema % host_info
```

Comment:
Shoudn't we keep a test to make sure '%s' is still working with this version ?

Reply:
Good point, I reverted one of the places where `%s` was used and added a `compat200` marker.